stdlib/format.ml

```diff
@@ -320,13 +320,17 @@ let format_pp_token state size = function
     | [] -> () (* No more tabulation block to close. *)
     end
 
-  | Pp_stab ->
+  | Pp_stab i ->
     begin match state.pp_tbox_stack with
     | Pp_tbox tabs :: _ ->
       let rec add_tab n = function
         | [] -> [n]
         | x :: l as ls -> if n < x then n :: ls else x :: add_tab n l in
-      tabs := add_tab (state.pp_margin - state.pp_space_left) !tabs
+      let i = match i with
+        | Some i when i <= state.pp_margin -> i
+        | _ -> state.pp_margin - state.pp_space_left
```

Comment:
you should use `None` rather than `_` (universal patterns at the toplevel of clauses are bad style).